src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -1126,9 +1130,34 @@ export class EditDataGridPanel extends GridParentComponent {
 		});
 	}
 
-	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): void {
+	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): boolean {
+		let result = true;
 		this.logService.debug('onBeforeEditCell called with grid: ' + event.grid + ' row: ' + event.row
 			+ ' cell: ' + event.cell + ' item: ' + event.item + ' column: ' + event.column);
+
+		let itemToEdit = event.item[event.cell].displayValue;
+
+		if (Services.DBCellValue.isDBCellValue(itemToEdit)) {
+			result = !this.hasNullAndLinebreak(itemToEdit.displayValue)
+		}
+		else if (typeof itemToEdit === 'string' || (itemToEdit && itemToEdit.text)) {
+			if (itemToEdit.text) {
+				result = !this.hasNullAndLinebreak(itemToEdit.text);
+			} else {
+				result = !this.hasNullAndLinebreak(itemToEdit);
+			}
+		}
+
+		if (!result) {
+			this.notificationService.warn(cellWithNullCharMessage);
+		}
+
+		return result;
+	}
+
+	private hasNullAndLinebreak(inputString: string): boolean {
+		let linebreakMatch = inputString.match(/(\r\n|\n|\r)/g);
```

Comment:
please do null check.

Reply:
also, I think you can remove the `/g` option, the match should return as soon as the first match is found, no need to find all the matches.